nerfstudio/data/dataparsers/sdfstudio_dataparser.py

```diff
@@ -116,17 +111,21 @@ def get_semantics(image_idx: int, semantic_path, group_path):
     # semantics
     semantic = np.array(Image.open(semantic_path[image_idx]), dtype="uint8")
     semantic = torch.from_numpy(semantic).type(torch.uint8)
+    results = {"semantics": semantic}
+    
     # groups for patch loss
-    groups_raw = np.array(Image.open(group_path[image_idx]), dtype="uint8")
-    groups_raw = torch.from_numpy(groups_raw).type(torch.uint8)
-    # make sure groups are unique and start from 0
-    groups = torch.zeros_like(groups_raw)
-    for i, group_id in enumerate(groups_raw.unique()):
-        groups[groups_raw == group_id] = i
-    groups = groups.type(torch.uint8)
+    if group_path[image_idx] is not None:
```

Comment:
this fixes the error, but it will make it hard to debug because the rest of the code now does not know whether the data includes masks or not.  
I think I would prefer a more explicit version where you add an option to the config, e.g. "include_groups" that just defaults to false

Reply:
My latest commit switch to this approach, with a new config `include_semantic_groups`.